src/vivarium_public_health/dataset_manager/artifact.py

```diff
@@ -33,9 +34,16 @@ def __init__(self, path: str, filter_terms: List[str]=None):
 
         self.path = path
         self._filter_terms = filter_terms
-
         self._cache = {}
-        self._keys = [EntityKey(k) for k in hdf.get_keys(self.path)]
+        self._keys = [EntityKey(k) for k in self._get_keys()]
+
```

Comment:
not pretty, but it guarantees that we have a valid file path and `meatada.keyspace` inside. better idea is always welcomed